local/rest_api_gcbm/app.py

```diff
@@ -177,71 +183,419 @@ def gcbm_upload():
+        data["Providers"]["RasterTiled"]["tileLonSize"] = tileLon
+
+        json.dump(data, provider_config, indent=4)
+        provider_config.truncate()
+
+        dictionary = {
+            "layer_type": "GridLayer",
+            "layer_data": "Byte",
+            "nodata": nd,
+            "tileLatSize": tileLat,
+            "tileLonSize": tileLon,
+            "blockLatSize": blockLat,
+            "blockLonSize": blockLon,
+            "cellLatSize": cellLat,
+            "cellLonSize": cellLon,
+        }
+
+        # should be able to accept variable number of inputs, but requires
+        # means for user to specify/verify correct ["attributes"]
+        def get_input_layers():
+            for root, dirs, files in os.walk(
+                os.path.abspath(f"{input_dir}/miscellaneous/")
+            ):
+                for file in files:
+                    fp2 = os.path.join(root, file)
+                    Rastersm.append(fp2)
+
+                for i in Rastersm:
+                    img = rst.open(i)
+                    d = img.nodata
+                    nodatam.append(d)
+
+            with open(
+                f"{input_dir}/initial_age_moja.json", "w", encoding="utf8"
+            ) as json_file:
+                dictionary["layer_type"] = "GridLayer"
+                dictionary["layer_data"] = "Int16"
+                dictionary["nodata"] = nodatam[1]
+                json.dump(dictionary, json_file, indent=4)
+
+            with open(
+                f"{input_dir}/mean_annual_temperature_moja.json", "w", encoding="utf8"
+            ) as json_file:
+                dictionary["layer_type"] = "GridLayer"
+                dictionary["layer_data"] = "Float32"
+                dictionary["nodata"] = nodatam[0]
+                json.dump(dictionary, json_file, indent=4)
+
+            with open(
+                f"{input_dir}/Classifier1_moja.json", "w", encoding="utf8"
+            ) as json_file:
+                dictionary["layer_type"] = "GridLayer"
+                dictionary["layer_data"] = "Byte"
+                dictionary["nodata"] = nd
+                dictionary["attributes"] = {
+                    "1": "TA",
+                    "2": "BP",
+                    "3": "BS",
+                    "4": "JP",
+                    "5": "WS",
+                    "6": "WB",
+                    "7": "BF",
+                    "8": "GA",
+                }
+                json.dump(dictionary, json_file, indent=4)
+
+            with open(
+                f"{input_dir}/Classifier2_moja.json", "w", encoding="utf8"
+            ) as json_file:
+                dictionary["layer_type"] = "GridLayer"
+                dictionary["layer_data"] = "Byte"
+                dictionary["nodata"] = nd
+                dictionary["attributes"] = {"1": "5", "2": "6", "3": "7", "4": "8"}
+                json.dump(dictionary, json_file, indent=4)
+
+            with open(
```

Comment:
Later we'll need to resolve how input files are matched to the right year (currently hard coded to match by filename). This is fine for now